backend/capellacollab/users/routes.py

```diff
@@ -27,68 +37,82 @@
     "/",
     response_model=t.List[User],
     responses=AUTHENTICATION_RESPONSES,
+    dependencies=[Depends(RoleVerification(required_role=Role.ADMIN))],
 )
-def get_users(token=Depends(JWTBearer()), db: Session = Depends(get_db)):
-    verify_admin(token, db)
+def get_users(
+    db: Session = Depends(get_db),
+):
     return users.get_all_users(db)
 
 
-@router.post("/", response_model=User, responses=AUTHENTICATION_RESPONSES)
-def create_user(token=Depends(JWTBearer()), db: Session = Depends(get_db)):
-    return users.create_user(db, get_username(token))
+@router.post(
+    "/",
+    response_model=User,
+    responses=AUTHENTICATION_RESPONSES,
+    dependencies=[Depends(RoleVerification(required_role=Role.ADMIN))],
+)
+def create_user(body: BaseUser, db: Session = Depends(get_db)):
+    return users.create_user(db, body.name)
 
 
 @router.get(
-    "/{username}",
+    "/current",
     response_model=User,
     responses=AUTHENTICATION_RESPONSES,
+    dependencies=[Depends(RoleVerification(required_role=Role.USER))],
 )
-def get_user(
-    username: str, db: Session = Depends(get_db), token=Depends(JWTBearer())
+def get_current_user(
+    db: Session = Depends(get_db), token=Depends(JWTBearer())
 ):
-    if username != get_username(token) and not is_admin(token, db):
-        raise HTTPException(
-            status_code=403,
-            detail={
-                "reason": "The username does not match with your username. You have to be administrator to see other users."
-            },
-        )
-    return users.get_user(db=db, username=username)
+    return users.get_user(db=db, username=get_username(token))
+
+
+@router.get(
+    "/{user_id}",
+    response_model=User,
+    responses=AUTHENTICATION_RESPONSES,
+    dependencies=[Depends(RoleVerification(required_role=Role.ADMIN))],
+)
+def get_user(user_id: int, db: Session = Depends(get_db)):
+    return users.get_user_by_id(db=db, user_id=user_id)
 
 
 @router.delete(
-    "/{username}", status_code=204, responses=AUTHENTICATION_RESPONSES
+    "/{user_id}",
+    status_code=204,
+    responses=AUTHENTICATION_RESPONSES,
+    dependencies=[Depends(RoleVerification(required_role=Role.ADMIN))],
 )
-def delete_user(
-    username: str, db: Session = Depends(get_db), token=Depends(JWTBearer())
-):
-    verify_admin(token, db)
-    project_users.delete_all_projects_for_user(db, username)
-    users.delete_user(db=db, username=username)
+def delete_user(user_id: str, db: Session = Depends(get_db)):
+    project_users.delete_all_projects_for_user(db, user_id)
+    users.delete_user(db=db, user_id=user_id)
```

Comment:
Personally, I like the approach with named parameters because it makes rebasing simpler, e.g. adding another optional parameter to the function. What's your opinion?

Reply:
For me it kind of depends on the type and context of the parameters. Using named parameters for required parameters, in my opinion, reduces the readability of the code, as it should then be applied consistently to all functions. However, I agree with you that using named parameters for optional parameters makes sense and should be used. In particular, this makes it easier to distinguish between required and optional parameters without going into the actual function.